src/pkgcore/ebuild/misc.py

```diff
@@ -66,8 +66,14 @@ def optimize_incrementals(sequence):
                 finalized.add(item)
 
 
-def incremental_chunked(orig, iterables):
+def incremental_chunked(orig: set[str], iterables):
     for cinst in iterables:
+        if "*" in cinst.neg:  # remove all previous set flags
+            orig.clear()
+        for flag in cinst.neg:
+            if flag.endswith("_*"):  # remove previous USE_EXPAND
```

Comment:
I'm surprised this is safe; ChunkedDataDict wasn't written as a use specific thing.  In tracing the code, it doesn't look like anything other than 'USE' pathways use the chunkdatadict however.